src/interfaces/ws.js

```diff
@@ -288,6 +296,20 @@ module.exports = function(app) {
     )
   }
 
+  function processReuestQuery(spark, msg) {
+    queryRequest(msg.requestId)
+      .then(reply => {
+        spark.write(reply)
+      })
+      .catch(err => {
+        spark.write({
+          requestId: msg.requestId,
+          state: 'COMPLETED',
```

Comment:
From 1on1 discussion in Slack: if there is no request found for the id `state` should be `undefined` as it refers to the original request, whose status we are querying here.

Furthermore 404 statusCode is not included in the possible return values for request-response over ws, so should be added.